From 0947da68bcd5a55533f359247e7a1f3a26234a1a Mon Sep 17 00:00:00 2001
From: Viorel Suman <viorel.suman@intel.com>
Date: Thu, 19 Feb 2015 15:40:34 +0200
Subject: [PATCH] iio: inv_mpu6050: Guard from going into "negative" territory

The patch blocks powering off of an already
powered off device by introducing one additional
check of the powerup counter.

Change-Id: I92bf98c92aabb1eba5cc8c474db285fdb9e5bc65
Tracked-On: https://jira01.devtools.intel.com/browse/IRDA-3810
Signed-off-by: Viorel Suman <viorel.suman@intel.com>
---
 drivers/iio/imu/inv_mpu6050/inv_mpu_core.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/iio/imu/inv_mpu6050/inv_mpu_core.c b/drivers/iio/imu/inv_mpu6050/inv_mpu_core.c
index 504d6d3..01c2323 100644
--- a/drivers/iio/imu/inv_mpu6050/inv_mpu_core.c
+++ b/drivers/iio/imu/inv_mpu6050/inv_mpu_core.c
@@ -169,11 +169,14 @@ int inv_mpu6050_set_power_itg(struct inv_mpu6050_state *st, bool power_on)
 						       0);
 		if (!result)
 			st->powerup_count++;
-	} else {
+	} else if (st->powerup_count > 0) {
 		st->powerup_count--;
 		if (!st->powerup_count)
 			result = inv_mpu6050_write_reg(st, st->reg->pwr_mgmt_1,
 						       INV_MPU6050_BIT_SLEEP);
+	} else {
+		dev_err(&st->client->dev,
+			"Attempt to power off an already powered off device\n");
 	}
 
 	if (result)
-- 
1.9.1

